File: backend/src/main/java/com/warlocksai/videoclub/model/Pelicula.java
package com.warlocksai.videoclub.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "peliculas")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Pelicula {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    @Column(nullable = false)
    private String titulo;

    @NotBlank
    private String director;

    @NotNull
    private Integer anio;

    @NotBlank
    private String genero;

    @Column(length = 1000)
    private String sinopsis;

    private String portadaUrl;

    private Boolean disponible = true;

    @OneToMany(mappedBy = "pelicula", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    @Builder.Default
    private List<Alquiler> alquileres = new ArrayList<>();
}

File: frontend/src/app/models/pelicula.ts
export interface Pelicula {
  id: number;
  titulo: string;
  director: string;
  directorId: number;
  directorNombre: string;
  anio: number;
  disponible: boolean;
  categoriaId: number;
  categoriaNombre: string;
}

export interface CrearPelicula {
  titulo: string;
  directorId: number;
  anio: number;
  disponible: boolean;
  categoriaId: number;
}
